src/link/Wasm.zig

```diff
@@ -309,17 +316,44 @@ fn linkWithLLD(self: *Wasm, comp: *Compilation) !void {
         _ = try man.hit();
         digest = man.final();
 
-        var prev_digest_buf: [digest.len]u8 = undefined;
-        const prev_digest: []u8 = Cache.readSmallFile(
+        var prev_id_data_buf: [255]u8 = undefined;
+        const prev_id_data = Cache.readSmallFile(
             directory.handle,
             id_symlink_basename,
-            &prev_digest_buf,
+            &prev_id_data_buf,
         ) catch |err| blk: {
             log.debug("WASM LLD new_digest={} error: {}", .{ digest, @errorName(err) });
             // Handle this as a cache miss.
-            break :blk prev_digest_buf[0..0];
+            break :blk prev_id_data_buf[0..0];
         };
-        if (mem.eql(u8, prev_digest, &digest)) {
+        const prev_digest = if (prev_id_data.len == 0)
+            prev_id_data
+        else
+            prev_id_data[0..digest.len];
+
+        if (mem.eql(u8, prev_digest, &digest)) blk: {
+            if (!mem.eql(u8, prev_id_data[digest.len..], sub_path)) {
+                // Rename the output to the new basename
+                directory.handle.rename(prev_id_data[digest.len..], sub_path) catch |err| {
+                    // If the rename fails, handle this as a cache miss.
+                    log.debug("Error renaming MachO linking output: {}", .{@errorName(err)});
```

Comment:
There's got to be a better way than copying this code four times.
`MachO` -> `Wasm`

Reply:
Thanks, will fix in a minute.  
I agree, I want to unify those somehow in another PR.